ksql-engine/src/main/java/io/confluent/ksql/function/udf/datetime/TimestampToString.java

```diff
@@ -16,45 +16,80 @@
 
 package io.confluent.ksql.function.udf.datetime;
 
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
 import io.confluent.ksql.function.KsqlFunctionException;
-import io.confluent.ksql.function.udf.Kudf;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
 import java.sql.Timestamp;
 import java.time.ZoneId;
 import java.time.format.DateTimeFormatter;
+import java.util.concurrent.ExecutionException;
 
-public class TimestampToString implements Kudf {
+@UdfDescription(name = "timestamptostring", author = "Confluent",
+    description = "Converts a BIGINT millisecond timestamp value into"
+        + " the string representation of the timestamp in the given format.")
+public class TimestampToString {
 
-  private DateTimeFormatter threadSafeFormatter;
+  private final LoadingCache<String, DateTimeFormatter> formatters =
+      CacheBuilder.newBuilder()
+          .maximumSize(1000)
+          .build(CacheLoader.from(DateTimeFormatter::ofPattern));
 
-  @Override
-  public Object evaluate(final Object... args) {
-    if (args.length < 2) {
-      throw new KsqlFunctionException("TimestampToString udf should have at least "
-                                    + "two input arguments: date value and format.");
+  @Udf(description = "Converts a BIGINT millisecond timestamp value into the"
+      + " string representation of the timestamp in the given format. Single quotes in the"
+      + " timestamp format can be escaped with '', for example: 'yyyy-MM-dd''T''HH:mm:ssX'"
+      + " The format pattern should be in the format expected"
+      + " by java.time.format.DateTimeFormatter")
+  public String timestampToString(
+      @UdfParameter(value = "epochMilli",
+          description = "Milliseconds since"
+              + " January 1, 1970, 00:00:00 GMT.") final long epochMilli,
+      @UdfParameter(value = "formatPattern",
+          description = "The format pattern should be in the format expected by"
+              + " java.time.format.DateTimeFormatter.") final String formatPattern) {
+    try {
+      final Timestamp timestamp = new Timestamp(epochMilli);
+      final DateTimeFormatter formatter = formatters.get(formatPattern);
+      return timestamp.toInstant()
+          .atZone(ZoneId.systemDefault())
```

Comment:
Do we want this to use the system timezone? i.e., dates are all UTC in KSQL. So this is deviating from the norm. if this is indeed the case that we want this, then can you update the description etc to make it clear what is happening?

Reply:
The old `StringToTimestamp` implementation used system default time zone when no time zone was provided. I can change to UTC and update the docs, but wouldn't this be considered a breaking change?

Reply:
I've assumed we want to keep the old behaviour for now and have updated the docs accordingly.